src/mtcp/mtcp_restart.c

```diff
@@ -1279,6 +1285,28 @@ hasOverlappingMapping(VA addr, size_t size)
   return ret;
 }
 
+NO_OPTIMIZE
+static int
+mremap_move(void *dest, void *src, size_t size) {
+  int mtcp_sys_errno;
+  if (dest == src) {
+    return 0; // Success
+  }
+  void *rc = mtcp_sys_mremap(src, size, size, MREMAP_FIXED | MREMAP_MAYMOVE,
+                               dest);
+  if (rc == dest) {
+    return 0; // Success
+  } else if (rc == MAP_FAILED) {
+    MTCP_PRINTF("***Error: failed to mremap; errno: %d.\n",
+                mtcp_sys_errno);
+    return -1; // Error
+  } else {
+    // We failed to move the memory to 'dest'.  Undo it.
+    mremap_move(src, rc, size);
```

Comment:
@gc00 could you explain what would be the workflow after we undo the mremap_move?
Wouldn't the vvar/vdso will still have overlapping areas?
If we need to abort after undo-ing then why do we need to undo mremap_move in the first place?